services/worker-manager/src/providers/google.js

```diff
@@ -413,11 +404,15 @@ class GoogleProvider extends Provider {
       if (worker.providerData.operation) {
         // We only check in on the operation if the worker failed to
         // start succesfully
-        await this.handleOperation({
+        if (await this.handleOperation({
           op: worker.providerData.operation,
           errors: this.errors[worker.workerPoolId],
           monitor,
-        });
+        })) {
+          monitor.debug('operation still running');
```

Comment:
I wonder if google can have operations that never complete? Maybe we set a max-retries to check in before we just mark it as failed?

Reply:
https://cloud.google.com/compute/docs/api/how-tos/api-requests-responses#%20%20%20%20%20%20%20%20%20%20handling_api_responses

> Your request is not complete until the Operation resource's status returns as DONE. This can take some time depending on the nature of your request. Then, after the operation's status returns as DONE, you can check to see if the operation succeeded and whether there were any errors.

That suggests strongly that it will eventually be DONE, so I'm inclined to assume that's true and if we find out otherwise we can figure out if it's a Google bug or a provider bug.  Since they suggest waiting for multiples of 2 minutes with WAIT, I'm not even sure what we'd set as a max wait time.